common/crypto/platform/openssl/crypto.cpp

```diff
@@ -93,6 +93,45 @@ string GetOpenSSLErrorMessage() {
 	return errorDescription;
 }
 
+// We need our own custom callback routine, as the default one will prompt for a
+// passphrase.
+int PrivateKey::CapturePrivateKeyPassphrase(char *buf, int size, int rwflag, void *u) {
+	// We'll only use this callback for reading passphrases, not for writing
+	// them.
+	assert(rwflag == 0); 
+
+	// Copy passphrase to out, out is not expected to be null terminated.
+	char *pass = (char *)u;
+	strncpy(buf, pass, size);
+	return (static_cast<int>(strlen(pass)) < size) ? static_cast<int>(strlen(pass)) : size;
+}
+
+ExpectedPrivateKey PrivateKey::LoadFromPEM(const string &private_key_path, const string &passphrase) {
+	auto private_bio_key = unique_ptr<BIO, void (*)(BIO *)>(
+		BIO_new_file(private_key_path.c_str(), "r"), bio_free_func);
+	if (!private_bio_key.get()) {
```

Comment:
In general we try to avoid `.get()` because it sidesteps the safety of unique_ptr when debugging is turned on. In this particular case, it doesn't matter, because it's just a boolean test, but it would matter for example here:

```c++
void myfunc(char &c);
...
unique_ptr<char> p;
// Passes invalid reference, might crash sometime long after.
myfunc(*p.get());
// Crashes immediately.
myfunc(*p);
```

So I think we should just stay away from it unless we are forced to use it.